cardano-node/src/Cardano/Node/Protocol/Shelley.hs

```diff
@@ -174,14 +231,15 @@ readLeaderCredentials (Just ProtocolFilepaths {
                tpraosLeaderCredentialsInitSignKey = kesKey,
                tpraosLeaderCredentialsLabel = "Shelley"
              }
-
--- But not OK to supply some of the files without the others.
-readLeaderCredentials (Just ProtocolFilepaths {shelleyCertFile = Nothing}) =
-    throwError OCertNotSpecified
-readLeaderCredentials (Just ProtocolFilepaths {shelleyVRFFile = Nothing}) =
-    throwError VRFKeyNotSpecified
-readLeaderCredentials (Just ProtocolFilepaths {shelleyKESFile = Nothing}) =
-    throwError KESKeyNotSpecified
+  where
+    parseEnvelope ::
```

Comment:
You shouldn't need `parseEnvelope` as we have `readFileTextEnvelope` in `cardano-api` already. You can lift that into `ExceptT`.

Reply:
there isn't quite a match there:

1. The code depends on `readEnvelope` not decoding the envelope which `readFileTextEnvelope` does:
```
   readEnvelope :: FilePath -> ExceptT ShelleyProtocolInstantiationError IO
                                       (TextEnvelope, FilePath)
   readEnvelope fp = do
     content <- handleIOExceptT (CredentialsReadError fp) $
                  BS.readFile fp
     firstExceptT (EnvelopeParseError fp) $ hoistEither $
       (, fp) <$> Aeson.eitherDecodeStrict' content
```
2. `readFileTextEnvelope` does `runExceptT`, pushes the computation outside of the `ExceptT` monad, which means it'll have to be rewrapped back into `ExceptT` by the code.. which is suboptimal.

Reply:
@deepfire Please provide a reasonable alternative and let @Jimbo4350 chose between them.

Reply:
`readFileTextEnvelope` reads the file and expects it to conform to a particular format.

Bulk credentials files a list of content triplets corresponding to what the usual credentials files contain, individually.

That means that `readFileTextEnvelope` cannot be used -- at all.

So there is no reasonable alternative.

cc @intricate @Jimbo4350 @erikd